pandas/core/frame.py

```diff
@@ -174,6 +174,18 @@
 
     .. versionadded:: 0.17.0
 
+validate : {None, '1:1', '1:m', 'm:1', 'm:m', "one_to_one", "one_to_many",
+            "many_to_one", "many_to_many"}, default None
```

Comment:
@jorisvandenbossche This relates to the question at the top of this PR: should we check the manys? In other words, if someone passed "one_to_many" but the merge is actually "one_to_one", is that ok? 

My sense is that's fine -- i've never been in a situation where getting 1:1 where I expected 1:m was a problem -- but one corollary is that "many_to_many" doesn't actually do anything. I like the idea of leaving it there -- it's nice to be able to label ones merges consistently in this way -- but that specification doesn't imply any tests (which is why I left it off the list).